### ISSUE TITLE
Viewing a profile hangs the browser

### ISSUE THREAD

**Issue by juj** [BabylonJS/Spector.js#151]
STR:

Capture a profile of

http://clb.confined.space/dump/PolyStyleMedievalVillage_20200418_065906_wasm_release_cpuprofiler/

It is a bit difficult to do due to bug #150, but once one manages to do a capture, there is something about viewing the capture that makes Spector.js hang altogether:

http://clb.confined.space/dump/SpectorJsHangsWhenViewingProfile.mov

Waiting for the UI to become responsive does not work, but the hang keeps occurring.

**Comment by sebavan** [comment]
We have 
![image](https://user-images.githubusercontent.com/16655165/79640169-6223ab80-8190-11ea-9874-8f0eb1052d40.png)

Actually the entire UI has been generated for a number of calls around 5000 This is amazingly rare to have that many commands per frame. I guess it also explains the hanging issue during the capture.

I am not sure If we should support that many calls per frame or not as it sounds really edge case to me regarding most of the projects I have seen so far.

So you think there might be more and more of those experiences around ???

**Comment by juj** [comment]
> This is amazingly rare to have that many commands per frame.

Yeah, agreed, the page is really heavy with WebGL. However, that is partly the reason why I am looking at it with a WebGL inspector in the first place!

Though it seems to me that this bug report is still valid: why should scrolling the API call list view, or clicking on an item in that list be O(n) to the number of items on the list? It seems that some work is being done that is linear to the number of recorded API calls?

Perhaps there is some state change accumulation tracking taking place? In that case, I'd understand that clicking on an item would be proportional to the number of api calls that took place before the clicked item, but the api calls that come afterwards should not have an effect?

**Comment by sebavan** [comment]
Definitely, it is just me getting stupid when i built the page :-)

What i can propose you if you have a bit of time is to fix it around may the 10th ? I ll have a bit of days off so I should be able to focus on these and the offscreen canvas i would like to upport.

**Comment by juj** [comment]
Cool, no rush at all! Help is much appreciated!

**Comment by sebavan** [comment]
The react move helped a tiny bit here. The UI is dead slow but usable... Closing for now but I ll reopen if we spot more of those.
